significant warnings on 3.6.1  and R-devel 2019-12-21 r77609

Found the following significant warnings:

  epanet.c:2472:12: warning: ignoring return value of ‘fread’, declared with attribute warn_unused_result [-Wunused-result]

  epanet.c:2474:12: warning: ignoring return value of ‘fread’, declared with attribute warn_unused_result [-Wunused-result]

  output.c:597:18: warning: ignoring return value of ‘fread’, declared with attribute warn_unused_result [-Wunused-result]

  report.c:448:15: warning: ignoring return value of ‘fread’, declared with attribute warn_unused_result [-Wunused-result]

  report.c:453:15: warning: ignoring return value of ‘fread’, declared with attribute warn_unused_result [-Wunused-result]

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -1,7 +1,7 @@
 Package: epanet2toolkit
 Type: Package
 Title: Call 'EPANET' Functions to Simulate Pipe Networks
-Version: 0.4.0
+Version: 0.4.1
 Date: 2019-12-30
 Authors@R: c(
    person("Ernesto", "Arandia", email="earandia.ie@gmail.com", role='aut'),

--- a/src/epanet.c
+++ b/src/epanet.c
@@ -2469,9 +2469,9 @@ int  openhydfile()
 /* the current network.                                  */
    if (Hydflag == USE)
    {
-      fread(&magic,sizeof(INT4),1,HydFile);
+      size_t res = fread(&magic,sizeof(INT4),1,HydFile);
       if (magic != MAGICNUMBER) return(306);
-      fread(&version,sizeof(INT4),1,HydFile);
+      res = fread(&version,sizeof(INT4),1,HydFile);
       if (version != ENGINE_VERSION) return(306);
       if (fread(nsize,sizeof(INT4),6,HydFile) < 6) return(306);
       if (nsize[0] != Nnodes  || nsize[1] != Nlinks ||

--- a/src/output.c
+++ b/src/output.c
@@ -594,7 +594,7 @@ int  savetimestat(REAL4 *x, char objtype)
          {
 
             /* Get output results for time period & update stats */
-            fread(x+1, sizeof(REAL4), n, TmpOutFile);
+            size_t res = fread(x+1, sizeof(REAL4), n, TmpOutFile);
             for (i=1; i<=n; i++)
             {
                xx = x[i];

--- a/src/report.c
+++ b/src/report.c
@@ -445,12 +445,12 @@ int  writeresults()
       /* Read in node results & write node table. */
       /* (Remember to offset x[j] by 1 because array is zero-based). */
       for (j=DEMAND; j<=QUALITY; j++)
-         fread((x[j-DEMAND])+1,sizeof(REAL4),Nnodes,OutFile);
+         size_t res = fread((x[j-DEMAND])+1,sizeof(REAL4),Nnodes,OutFile);
       if (nnv > 0 && Nodeflag > 0) writenodetable(x);
 
       /* Read in link results & write link table. */
       for (j=FLOW; j<=FRICTION; j++)
-         fread((x[j-FLOW])+1,sizeof(REAL4),Nlinks,OutFile);
+         size_t res2 = fread((x[j-FLOW])+1,sizeof(REAL4),Nlinks,OutFile);
       if (nlv > 0 && Linkflag > 0) writelinktable(x);
       Htime += Rstep;
    }

--- a/src/report.c
+++ b/src/report.c
@@ -438,6 +438,7 @@ int  writeresults()
    fseek(OutFile,OutOffset2,SEEK_SET);
    Htime = Rstart;
 
+   size_t res;
    /* For each reporting time: */
    for (np=1; np<=Nperiods; np++)
    {
@@ -445,12 +446,12 @@ int  writeresults()
       /* Read in node results & write node table. */
       /* (Remember to offset x[j] by 1 because array is zero-based). */
       for (j=DEMAND; j<=QUALITY; j++)
-         size_t res = fread((x[j-DEMAND])+1,sizeof(REAL4),Nnodes,OutFile);
+         res = fread((x[j-DEMAND])+1,sizeof(REAL4),Nnodes,OutFile);
       if (nnv > 0 && Nodeflag > 0) writenodetable(x);
 
       /* Read in link results & write link table. */
       for (j=FLOW; j<=FRICTION; j++)
-         size_t res2 = fread((x[j-FLOW])+1,sizeof(REAL4),Nlinks,OutFile);
+         res = fread((x[j-FLOW])+1,sizeof(REAL4),Nlinks,OutFile);
       if (nlv > 0 && Linkflag > 0) writelinktable(x);
       Htime += Rstep;
    }

--- a/tests/testthat/test_epanet.r
+++ b/tests/testthat/test_epanet.r
@@ -9,17 +9,24 @@
 context("ENepanet")
 test_that("ENepanet all args",{
   expect_silent( ENepanet("Net1.inp", "Net1.rpt", "Net1.bin") ) 
+  # clean-up 
+  file.remove("Net1.rpt") 
+  file.remove("Net1.bin") 
 })
 
 test_that("RENepanet null bin",{
 
   expect_silent( ENepanet("Net1.inp", "Net1.rpt") ) 
+  # clean-up 
+  file.remove("Net1.rpt") 
 })
 
 test_that("retuns null invisibly",{
 	 wv <- withVisible( ENepanet("Net1.inp", "Net1.rpt") ) 
 	 expect_null(wv$value)
 	 expect_false(wv$visible)
+  # clean-up 
+  file.remove("Net1.rpt") 
  }) 
 
 test_that("R func ENepanet bad input",{

